buildflow/io/clickhouse/strategies/clickhouse_strategies.py

```diff
@@ -0,0 +1,87 @@
+import asyncio
+import logging
+from typing import Any, Callable, Dict, Iterable, Type
+
+import clickhouse_connect
+import pandas as pd
+
+from buildflow.core.credentials import EmptyCredentials
+from buildflow.core.types.clickhouse_types import (
+    ClickhouseDatabase,
+    ClickhouseHost,
+    ClickhousePassword,
+    ClickhouseTableID,
+    ClickhouseUsername,
+)
+from buildflow.io.strategies.sink import SinkStrategy
+from buildflow.io.utils.schemas import converters
+
+_MAX_CONNECT_TRIES = 25
+
+
+class ClickhouseSink(SinkStrategy):
+    def __init__(
+        self,
+        *,
+        credentials: EmptyCredentials,
+        host: ClickhouseHost,
+        username: ClickhouseUsername,
+        password: ClickhousePassword,
+        database: ClickhouseDatabase,
+        table: ClickhouseTableID,
+    ):
+        super().__init__(credentials=credentials, strategy_id="clickhouse-sink")
+        self.client = None
+        self.database = database
+        self.table = table
+        self.host = host
+        self.username = username
+        self.password = password
+        try:
+            self.connect()
+        except ConnectionError:
```

Comment:
Should this raise the error instead of just swallowing it? That way it will fail fast